### PR TITLE
sc-60: Sc 60/create nft page

### PR DESCRIPTION
**Pull request by akshay-rakheja**
added pinata.js -> helps with pinning data to ipfs.

### CODE REVIEW

File: scripts/interact.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      return {
+        address: "",
+        status: "😥 " + err.message,
+      };
+    }
+  } else {
+    return {
+      address: "",
+      status: (
+        <span>
```
Comment (connie-ho) [inline scripts/interact.js:30]: Is this page suppose to be returning a react component? Where is this script used?

File: scripts/minter.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
+import { useEffect, useState } from "react";
```
Comment (connie-ho) [inline scripts/minter.js:1]: Should make a component called Minter.jsx

File: scripts/interact.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
+import { pinJSONToIPFS } from "./pinata.js";
```
Comment (connie-ho) [inline scripts/interact.js:1]: Move this to hooks/use-interact.js

File: scripts/pinata.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+require('dotenv').config();
```
Comment (connie-ho) [inline scripts/pinata.js:1]: usePinata

**Review by connie-ho — CHANGES_REQUESTED** [review]
move to react app folder

File: frontend/src/components/minter.jsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  useEffect(async () => {
+    const { address, status } = await checkWalletConnection();
+
+    setWallet(address);
+   // setStatus(status);
+
+    addWalletListener();
+  }, []);
+
+  function addWalletListener() {
```
Comment (connie-ho) [inline frontend/src/components/minter.jsx:31]: These functions could be extracted into a helper hook to clean up the component. It'll also help with testing (you can follow use-wallet-connection) as an example

File: frontend/src/hooks/use-interact.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
+import { pinJSONToIPFS } from "./use-pinata.js";
+require("dotenv").config();
+const alchemyKey = process.env.REACT_APP_ALCHEMY_KEY;
+const contractABI = require("../contract-abi.json"); // find where abi is generated and replace.
+const contractAddress = "0x4C4a07F737Bf57F6632B6CAB089B78f62385aCaE"; // need to change it to our contract address
```
Comment (connie-ho) [inline frontend/src/hooks/use-interact.js:5]: You can import this directly from the config folder

File: frontend/src/components/minter.jsx [hunk 7]
```diff
@@ -0,0 +1,122 @@
+import { useEffect, useState } from "react";
```
Comment (connie-ho) [inline frontend/src/components/minter.jsx:1]: small nit! Component files should be capitalized

File: frontend/src/components/minter.jsx [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+    //setStatus(status);
+    if (success) {
+      setName("");
+      setDescription("");
+      setURL("");
+    }
+  };
+
+  return (
+    <div className="Minter">
```
Comment (connie-ho) [inline frontend/src/components/minter.jsx:75]: Please use styled components or useStyles when possible plspsspslps <3

File: frontend/src/hooks/use-wallet-connection.js [hunk 9]
```diff
@@ -15,7 +15,8 @@ function useWalletConnection() {
     if (!loggedIn) return;
 
     //grab provider to get account
-    const provider = new ethers.providers.Web3Provider(window.ethereum, 'any');
+    //const provider = new ethers.providers.Web3Provider(window.ethereum, 'any');
+    const provider = new ethers.getDefaultProvider('ropsten'); //saw this implementation here:  https://docs.ethers.io/v4/api-providers.html -> not sure but this seems like the recommended way to do it.
```
Comment (connie-ho) [inline frontend/src/hooks/use-wallet-connection.js:19]: This is really good!!!! I think we can even change the ethers context with this now, but I can take this as a follow up

File: frontend/src/hooks/use-interact.js (hunk 6 above)
Comment (akshay-rakheja) [reply frontend/src/hooks/use-interact.js:5]: thanks was looking for this.

File: scripts/pinata.js (hunk 4 above)
Comment (akshay-rakheja) [reply scripts/pinata.js]: done

File: frontend/src/components/minter.jsx (hunk 5 above)
Comment (akshay-rakheja) [reply frontend/src/components/minter.jsx]: tried doing it. can you guys please check if its done right??

File: frontend/src/components/minter.jsx (hunk 7 above)
Comment (akshay-rakheja) [reply frontend/src/components/minter.jsx:1]: done

File: frontend/src/components/minter.jsx (hunk 8 above)
Comment (akshay-rakheja) [reply frontend/src/components/minter.jsx:70]: lol not sure what this is. will need your guys' help

**Comment by connie-ho** [comment]
good job with the hook! Just need to fix those build (linting) errors and a few other suggestions and you are good to merge ::::::::))))))

**Review by connie-ho — CHANGES_REQUESTED** [review]

File: frontend/src/components/minter.jsx (hunk 5 above)
Comment (connie-ho) [reply frontend/src/components/minter.jsx]: good job it looks pretty good to me!

**Pull request closed without merging**